### ISSUE TITLE
Not Working

### ISSUE THREAD

**Issue by scopertos** [Torantulino/Auto-GPT#620]
Continue (y/n): y
Using memory of type: LocalCache
Traceback (most recent call last):
  File "/Users/waiel/Auto-GPT/scripts/main.py", line 297, in <module>
    assistant_reply = chat.chat_with_ai(
                      ^^^^^^^^^^^^^^^^^^
  File "/Users/waiel/Auto-GPT/scripts/chat.py", line 72, in chat_with_ai
    relevant_memory = permanent_memory.get_relevant(str(full_message_history[-5:]), 10)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/waiel/Auto-GPT/scripts/memory/local.py", line 102, in get_relevant
    embedding = get_ada_embedding(text)
                ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/waiel/Auto-GPT/scripts/memory/base.py", line 9, in get_ada_embedding
    return openai.Embedding.create(input=[text], model="text-embedding-ada-002")["data"][0]["embedding"]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/openai/api_resources/embedding.py", line 33, in create
    response = super().create(*args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/openai/api_resources/abstract/engine_api_resource.py", line 149, in create
    ) = cls.__prepare_create_request(
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/openai/api_resources/abstract/engine_api_resource.py", line 106, in __prepare_create_request
    requestor = api_requestor.APIRequestor(
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/openai/api_requestor.py", line 130, in __init__
    self.api_key = key or util.default_api_key()
                          ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/openai/util.py", line 186, in default_api_key
    raise openai.error.AuthenticationError(
openai.error.AuthenticationError: No API key provided. You can set your API key in code using 'openai.api_key = <API-KEY>', or you can set the environment variable OPENAI_API_KEY=<API-KEY>). If your API key is stored in a file, you can point the openai module at it with 'openai.api_key_path = <PATH>'. You can generate API keys in the OpenAI web interface. See https://onboard.openai.com for details, or email support@openai.com if you have any questions.

**Comment by keenborder786** [comment]
Please set your API key in .env file. Refer to readme file.

**Comment by scopertos** [comment]
> Please set your API key in .env file. Refer to readme file.

I set it up in there twice, tried 2 different keys.

**Comment by keenborder786** [comment]
Can you please make sure that .env.template is renamed to the .env file? 
If you have already done that as well. Then, can you make sure that the API key has properly been given proper access?

**Comment by scopertos** [comment]
> Can you please make sure that .env.template is renamed to the .env file?
> If you have already done that as well. Then, can you make sure that the API key has properly been given proper access?

I already renamed it to just .env

My API should have full access and even access to GPT-4 32k Token system.

**Comment by keenborder786** [comment]
Can you please send me a screen shot of your .env file with the keys hidden. Thank you.

**Comment by scopertos** [comment]
<img width="977" alt="Screenshot 2023-04-09 at 2 12 35 PM" src="https://user-images.githubusercontent.com/129682753/230796781-caab950d-f792-444b-89b6-4ce88ad334db.png">

I took off the last 4 of the APIs just for the photo

**Comment by keenborder786** [comment]
Can you please change '= ' to '='(no space) when declaring OPENAI_API_KEY? This will fix your issue.

**Comment by hunterv3991** [comment]
Did that fix the issue? I'm having the exact same problem, but I do not have an extra space in the OPENAI_AI_KEY

**Comment by scopertos** [comment]
> Can you please change '= ' to '='(no space) when declaring OPENAI_API_KEY? This will fix your issue.

Still doesn't work, I tried that space only after I tried my second API key. It still doesn't work even without the space

**Comment by keenborder786** [comment]
Hmmm okay please let me have a look at this one then.

**Comment by keenborder786** [comment]
Can you please try one more thing:

- Delete all previous keys
- Generate new ones.

If this also does not work then please generate API keys from new account and see if that works.

**Comment by scopertos** [comment]
> Can you please try one more thing:
> 
> * Delete all previous keys
> * Generate new ones.
> 
> If this also does not work then please generate API keys from new account and see if that works.

Like I said I already tried generating a new key and it still didn't work

**Comment by comangallc** [comment]
I have the same issue I'm gonna give this tool 1-2 weeks its not working needs far more optimization to run, for now I will just use Jarvis for now or wait because the competition is here with Baby AGI. I respect what the creator is doing but the interest he is getting is moving focus away from optimizing because people I know are saying you need 12gb to run this & so on. Also he needs a detailed PDF & Video tutorial on the install process

**Comment by psikosen** [comment]
I had to jump back to the precious build pinecone messed a lot of things up for me, unfortunately  :(

**Comment by stefhugo** [comment]
Been having this same issue for the past 2 hours, hope your fixes are as simple as mine. It was literally a case of opening the .env file and selecting "save as" to make sure it does not save as a template file again but rather as environment (.env) file.

**Comment by SublimeAirPower** [comment]
That is the answer^^^^ so simple yet so important
